openlibrary/plugins/worksearch/code.py

```diff
@@ -1018,23 +1018,28 @@ def work_search(query, sort=None, page=1, offset=0, limit=100, fields='*', facet
     query['wt'] = 'json'
     if sort:
         sort = process_sort(sort)
+
+    if 'availability' in fields and 'ia' not in fields:
+        fields.append('ia')
     try:
-        (reply, solr_select, q_list) = run_solr_query(query,
-                                                      rows=limit,
-                                                      page=page,
-                                                      sort=sort,
-                                                      offset=offset,
-                                                      fields=fields,
-                                                      facet=facet,
-                                                      spellcheck_count=spellcheck_count)
+        (reply, solr_select, q_list) = run_solr_query(
+            query,
+            rows=limit,
+            page=page,
+            sort=sort,
+            offset=offset,
+            fields=fields,
+            facet=facet,
+            spellcheck_count=spellcheck_count
+        )
         response = json.loads(reply)['response'] or ''
     except (ValueError, IOError) as e:
         logger.error("Error in processing search API.")
         response = dict(start=0, numFound=0, docs=[], error=str(e))
 
     # backward compatibility
     response['num_found'] = response['numFound']
-    if fields == '*' or 'availability' in fields:
+    if '*' in fields or 'availability' in fields:
```

Comment:
This is the one piece I'm wary about for performance reasons; in order to get this out today, let's drop this piece. You can tell the person who emailed you that they need to change their query URLs to `fields=*,availability`, or, if they want smaller/faster/tidier, `fields=key,ia,availability`.

```suggestion
    if 'availability' in fields:
```